### ISSUE TITLE
Question: how to handle long running processes?

### ISSUE THREAD

**Issue by Christilut** [andywer/threads.js#317]
I'm just wondering what the recommended way is to tackle the following problem:

I have a long running external process (python) that waits for stdin and outputs on stdout. It doesn't restart every time because the overhead would be far longer than the execution time. I want to launch this process 16 times with `child_process.spawn`.

Now I'm not sure how I achieve this with Threads, because so far I've only used it with short running processes that exit when done.

I tried to create the process outside the scope in the file where my `expose` function is, but that ends up with all workers sharing the same child process globally.

Any suggestions?

**Comment by andywer** [comment]
Hmm, I'm not even sure threads.js is the right tool for the task you describe. threads.js purpose is to make it easy to run JS/TS code on multiple threads. `child_process.spawn()` is out of its scope and the question is: What benefit should threads.js provide in this scenario, given that the actual work is already delegated from the main JS thread to child processes.

I can only guess that you would like to be able to call your python program transparently using just function calls and maybe re-use the thread pool.

**Comment by Christilut** [comment]
Yeah exactly, I'm using the thread pool for other short tasks and it works great for that. That's why I was hoping to re-use it for these tasks too.

I'll do some experiments and see if I can figure something out.
